src/stdio-bridge/stdio-bridge.c

```diff
@@ -131,7 +151,11 @@ int main(int argc, char *argv[]) {
                 goto finish;
         }
 
-        r = sd_bus_set_address(a, arg_bus_path);
+        if (transport == BUS_TRANSPORT_MACHINE)
+                r = bus_set_address_system_machine(a, arg_bus_path);
+        else
+                r = sd_bus_set_address(a, arg_bus_path);
```

Comment:
Then this block can be simply
```
if (arg_bus_path)
 ...
```
Then, not necessary to crate new global variable `transport`.

Reply:
actually arg_bus_path gets set if the user provides it (for local transport)

Reply:
https://github.com/shawnl/systemd/blob/18a6ee5ff77c492ba982af3cf8ceb61e221b9160/src/stdio-bridge/stdio-bridge.c#L90

Reply:
Ouch! Please forget the comment...

Reply:
Well, you still caught a bug, that that was not malloced, as now required.